extension/apmconfigextension/opamp_callbacks.go

```diff
@@ -106,26 +130,26 @@ func (rc *remoteConfigCallbacks) onMessage(ctx context.Context, conn types.Conne
 	agentUidField := zap.String("instance_uid", agentUid)
 	if message.GetAgentDisconnect() != nil {
 		rc.logger.Info("Disconnecting the agent from the remote configuration service", agentUidField)
-		rc.agentState.Delete(agentUid)
+		_ = rc.agentState.Remove(agentUid)
```

Comment:
If i understand correctly, the fear is that disconnect messages may become lost and cache entries pile up (-> memory leak).

With the use of an LRU you limit the number of cache entries. So you limit the amount of leaked memory. That's a good step forward.
Ideally, you also have go routine that regularly calls PurgeExpired() to actively purge expired entries from the cache. A health check every 30s was mentioned - that might be a good place to call PurgeExpired(). The alternative is to start a small go routine dedicated for this task.

Reply:
Sounds like a good improvement to save some memory, purging cache based on the TTL https://github.com/elastic/opentelemetry-collector-components/pull/629/commits/13b77750303396e7b669fcb5da3030ea532e7384

wdyt?

Reply:
LGTM